test/ArrayFromString.ts

```diff
@@ -0,0 +1,35 @@
+import * as assert from 'assert'
+import { ArrayFromString, NumberFromString } from '../src'
+import { assertSuccess, assertFailure } from './helpers'
+
+describe('ArrayFromString', () => {
+  it('name', () => {
+    const T = ArrayFromString(NumberFromString, undefined, undefined, 'T')
+    assert.strictEqual(T.name, 'T')
+  })
+  it('is', () => {
+    const T = ArrayFromString(NumberFromString)
+    assert.strictEqual(T.is([1, 2]), true)
+    assert.strictEqual(T.is(['1', 2]), false)
+    assert.strictEqual(T.is('1,2'), false)
+  })
+  it('decode', () => {
+    const T = ArrayFromString(NumberFromString, /[\s,]+/)
+    assertSuccess(T.decode('1,2,3'), [1, 2, 3])
```

Comment:
It is worth denoting that the empty string decodes to an empty array. I've been using a similar codec that takes a NonEmptyString and returns a NonEmptyArray. I'm not sure what the use-case is for decoding an empty string with this codec but I suppose it can be piped to NEA